### PR TITLE
Create static webpage to get and display file content

### LINKED ISSUE #1: Create static webpage that takes file input and loads content in js

### PR DESCRIPTION
**Pull request by silvs110**
## Description
Created static page to get and display contents of a file.

Fixes #1

## Author's Checklist
- [x] **These changes don't break existing dependants.** If they do, and you suggest to proceed, 
please explain why.

## Reviewer's Checklist
- [ ] **All classes have documentation comments.** - NA
- [ ] **All methods have documentation comments.** - NA
- [ ] **No parameters are hardcoded.** Should be loaded as properties instead. If there needs to be hardcoded properties, explain why. - NA
- [ ] **All information to be logged/displayed to the user are internationalized.** If not, explain why. - NA
- [ ] **README.md has been updated if needed.** - NA
- [ ] **pom.xml version is set to the latest date** If not, explain why. - NA

## Assignee's Checklist
- [ ] **All GitHub action checks have passed.** - NA
- [x] **All reviewers have approved.**
- [ ] **Relevant documentation has been updated if needed.** - NA
- [ ] **pom.xml version is set to the latest date** If not, explain why. - NA

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+  <meta charset="utf-8">
+  <meta name="viewport" content="width=device-width, initial-scale=1">
+  <title>Sanitizer</title>
+
+  <meta name="theme-color" content="#fafafa">
+  <!-- Custom styles -->
+  <link href="style.css" rel="stylesheet" type="text/css" />
+
+  <script src="https://cdn.jsdelivr.net/npm/bootstrap@5.3.3/dist/js/bootstrap.bundle.min.js" integrity="sha384-YvpcrYf0tY3lHB60NNkmXc5s9fDVZLESaAA55NDzOxhy9GkcIdslK1eN7N6jIeHz" crossorigin="anonymous"></script>
+  <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.3.3/dist/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-QWTKZyjpPEjISv5WaRU9OFeRpok6YctnYmDr5pNlyT2bRjXh0JMhjY6hW+ALEwIH" crossorigin="anonymous">
```
Comment (the-c0d3br34k3r) [inline index.html:14]: Can we add the bootstrap files to the repo as well and refer to them using local paths for now?

File: index.html [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+</head>
+
+<body>
+
+  <!-- Add your site or application content here -->
+  <h1 id="title">HAR File Sanitizer</h1>
```
Comment (the-c0d3br34k3r) [inline index.html:20]:
```suggestion
  <h1 id="title">Sensitive Info Sanitizer</h1>
  <p>Currently this tool supports only HAR files.</p>
```

File: index.html [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+  <div id="uploader">
+    <input type="file" id="upload_button" name="upload_button" class="form-control" accept=".har"/>
+  </div>
+
+  <div id="display_panel">
+    <div id="display_card" class="card">
+      <div class="card-header" id="card_header">
+        Sanitized HAR file
```
Comment (the-c0d3br34k3r) [inline index.html:28]:
```suggestion
        Sanitized file
```

File: index.html [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+      </div>
+      <div class="card-body">
+          <div id="overlay-spinner">
+            <div class="w-100 d-flex justify-content-center align-items-center" id="spinner_container">
+              <div class="spinner-border text-success" id="spinner"></div>
+            </div>
+          </div>
+        <div class="scrollable" id="scrollable_content">
+          <p class="card-text" id="output_content">
```
Comment (the-c0d3br34k3r) [inline index.html:37]: (nitpick)
```suggestion
          <p class="card-text" id="output_content"></p>
```

**Review by the-c0d3br34k3r — CHANGES_REQUESTED** [review]
A clean look! Love it.

Just minor changes needed.

Aside the comments on the code, the following needs to be addressed:
- The download button overflows when the file is loaded.
   Screenshot 2024-02-21 at 8 30 36 PM
- It seems to not work in Firefox. Let's ensure that it works in both Chrome and Firefox.

File: index.html (hunk 1 above)
Comment (silvs110) [reply index.html:14]: I'm not following

File: index.html [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+          <input type="button" id="download_button" class="btn btn-success" name="download_button" value="Download File" disabled>
+      </div>
+    </div>
+  </div>
+
+  <script>
+    document.getElementById("upload_button").addEventListener("change", getFile);
+    document.getElementById("download_button").addEventListener("click",  downloadContent);
+
+    function getFile(event) {
```
Comment (the-c0d3br34k3r) [inline index.html:48]: Move the js code to a separate file.

File: index.html [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+      </div>
+      <div class="card-footer text-body-secondary" id="downloader">
+          <input type="button" id="download_button" class="btn btn-success" name="download_button" value="Download File" disabled>
+      </div>
+    </div>
+  </div>
+
+  <script>
+    document.getElementById("upload_button").addEventListener("change", getFile);
+    document.getElementById("download_button").addEventListener("click",  downloadContent);
```
Comment (the-c0d3br34k3r) [inline index.html:46]:
```suggestion
```

File: index.html [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+  <link href="css/style.css" rel="stylesheet" type="text/css" />
+
+  <script src="script/bootstrap.bundle.min.js"></script>
+  <link href="css/bootstrap.min.css" rel="stylesheet">
+</head>
+
+<body>
+  <h1 id="title">Sensitive Info Sanitizer</h1>
+  <div id="uploader">
+    <input type="file" id="upload_button" name="upload_button" class="form-control" accept=".har"/>
```
Comment (the-c0d3br34k3r) [inline index.html:20]:
```suggestion
    <input type="file" id="upload_button" name="upload_button" class="form-control" accept=".har" onchange="getFile(event)" />
```

File: index.html [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+            <div class="w-100 d-flex justify-content-center align-items-center" id="spinner_container">
+              <div class="spinner-border text-success" id="spinner"></div>
+            </div>
+          </div>
+        <div class="scrollable" id="scrollable_content">
+          <p class="card-text" id="output_content"></p>
+        </div>
+      </div>
+      <div class="card-footer text-body-secondary" id="downloader">
+          <input type="button" id="download_button" class="btn btn-success" name="download_button" value="Download File" disabled>
```
Comment (the-c0d3br34k3r) [inline index.html:39]:
```suggestion
          <input type="button" id="download_button" class="btn btn-success" name="download_button" value="Download File" onclick="downloadContent(event)" disabled>
```

**Review by the-c0d3br34k3r — CHANGES_REQUESTED** [review]
Thanks for the quick follow up!
Have requested more changes.

File: index.html (hunk 1 above)
Comment (the-c0d3br34k3r) [reply index.html]: Looks like you already did it i.e. include the bootstrap js and css as file sin the PR.

**Review by the-c0d3br34k3r — APPROVED** [review]

**Pull request merged**